torch/_dynamo/symbolic_convert.py

```diff
@@ -1086,10 +1086,22 @@ def run(self):
             except TensorifyScalarRestartAnalysis:
                 raise
             except BackendCompilerFailed:
+                raise
+            except RuntimeError as e:
+                if hasattr(e, "msg") and "Could not guard on data-dependent" in e.msg:
+                    print(
+                        "\n"
+                        + torch.fx.GraphModule({}, self.output.graph).print_readable(
+                            print_output=False, include_stride=True, include_device=True
+                        ),
+                        file=sys.stderr,
+                    )
+
```

Comment:
My vote is to not log-spam users. For example, cpython doesn't print the bytecode on exception; we've been working to get rid of the internal compiler stack trace from users on user error. It's also not completely clear to me what users will use the fx graph for.

This seems like something that's better hidden behind a config option. If we think that export users want this by default, then we could provide it by default for export.

Reply:
Agreed with all of the above. The other use case in my brain right now is that we have examples of OSS users who run into a bug but can't easily give us a repro (and may not want to give us a tlparse either, if it leaks model info).

So there is probably still use case for a "print verbose errors" envvar/config throughout PT2. Do you have any thoughts on:

(1) having that as a config at all
(2) mentioning the config in our default PT2 error message, so they know more about what to include if they want to submit a bug report (I'm not sure what the general plan is for our default error message today)

Reply:
Yes, we need a specific config flag for "if you're reporting a bug report to us, please set this config and give us the output". Since the POR is to hide the internal compiler stack trace from users, we'd need this config to actually show the internal compiler stack trace. Could be cool to dump other information there too (like the partial FX graph)

Reply:
Bob and I chatted a bit offline. The consensus was that printing partial graphs is useful for users to debug user errors with dynamic shapes (you can follow the symints in the graph), so we should just do it for the dynamic shapes case. It's fairly easy to turn off or try a different strategy if users tell us it's too spammy after partial fx graph printing is turned on.